restful.module

```diff
@@ -573,11 +573,11 @@ function restful_date_time_format_element_validate($element, &$form_state) {
 function restful_restful_resource_alter(ResourceInterface &$resource) {
   $plugin_definition = $resource->getPluginDefinition();
   // Check for the rate limit configuration.
-  if (!empty($plugin_definition['rateLimit'])) {
-    $resource = new RateLimitedResource($resource);
-  }
-  // Check for the rate limit configuration.
   if (!empty($plugin_definition['renderCache']['render'])) {
     $resource = new CachedResource($resource);
   }
+  // Check for the rate limit configuration.
+  if (!empty($plugin_definition['rateLimit'])) {
```

Comment:
Actually Cache needs to be the first one (the one more deep in the composition), since cache will shortcut and return when there is a cache hit.